frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -777,6 +779,112 @@
 
         };
 
+        vm.getAllSubmissionResults = function(phaseId) {
+
+            vm.stopFetchingSubmissions = function() {
+                $interval.cancel(vm.poller);
+            };
+
+            vm.stopFetchingSubmissions();
+            vm.isResult = true;
+            vm.phaseId = phaseId;
+
+            // loader for loading submissions.
+            vm.startLoader =  loaderService.startLoader;
+            vm.startLoader("Loading Submissions");
+
+            // get submissions of all the challenge phases
+            vm.isNext = '';
+            vm.isPrev = '';
+            vm.currentPage = '';
+            vm.showPagination = false;
+
+            var parameters = {};
+            parameters.url = "challenges/" + vm.challengeId + "/challenge_phase/" + vm.phaseId + "/submissions/";
+            parameters.method = 'GET';
+            parameters.data = {};
+            parameters.token = userKey;
+            parameters.callback = {
+                onSuccess: function(response) {
+                    var details = response.data;
+
+                    vm.submissionResult = details;
+
+                    if (vm.submissionResult.count === 0) {
+                        vm.showPagination = false;
+                        vm.paginationMsg = "No results found";
+                    } else {
+
+                        vm.showPagination = true;
+                        vm.paginationMsg = "";
+                    }
+
+                    if (vm.submissionResult.next === null) {
+                        vm.isNext = 'disabled';
+                    } else {
+                        vm.isNext = '';
+
+                    }
+                    if (vm.submissionResult.previous === null) {
+                        vm.isPrev = 'disabled';
+                    } else {
+                        vm.isPrev = '';
+                    }
+                    if (vm.submissionResult.next !== null) {
+                        vm.currentPage = vm.submissionResult.next.split('page=')[1] - 1;
+                    } else {
+                        vm.currentPage = 1;
+                    }
+
+                    vm.load = function(url) {
+                        // loader for loading submissions
+                        vm.startLoader =  loaderService.startLoader;
+                        vm.startLoader("Loading Submissions");
+                        if (url !== null) {
+
+                            //store the header data in a variable
+                            var headers = {
+                                'Authorization': "Token " + userKey
+                            };
+
+                            //Add headers with in your request
+                            $http.get(url, { headers: headers }).then(function(response) {
```

Comment:
This is bad way to make a request, please make a function that is called whenever a user wants to perform AJAX, you can make a custom service.
The controllers can call this service whenever we need an AJAX call.